hazelcast/src/test/java/com/hazelcast/concurrent/semaphore/SemaphoreBasicTest.java

```diff
@@ -103,6 +103,31 @@ public void testRelease() {
 
         assertEquals(semaphore.availablePermits(), numberOfPermits);
     }
+    
+    @Test(timeout = 30000)
+    public void testAllowNegativePermits() {
+        assertTrue(semaphore.init(10));
+        
+        semaphore.reducePermits(15);
+        
+        assertEquals(0, semaphore.availablePermits());
+        
+        semaphore.release(10);
+        
+        assertEquals(5, semaphore.availablePermits());
+    }    
+    
+    
+    @Test(timeout = 30000)
+    public void testIncreasePermits() {
```

Comment:
Could you please add a test case to verify that an `IllegalArgumentException` is thrown if the permit is negative?